src/main/java/uk/gov/companieshouse/mapper/DirectorApprovalMapper.java

```diff
@@ -0,0 +1,29 @@
+package uk.gov.companieshouse.mapper;
+
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.model.db.DirectorApproval;
+import uk.gov.companieshouse.model.db.DissolutionDirector;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Service
+public class DirectorApprovalMapper {
+    public DirectorApproval mapToDirectorApproval(String userId, String ip) {
+        final DirectorApproval approval = new DirectorApproval();
+        approval.setUserId(userId);
+        approval.setIpAddress(ip);
+        approval.setDateTime(LocalDateTime.now());
+        return approval;
+    }
+
+    public boolean mapDirectorsApprovalToBoolean(List<DissolutionDirector> dissolutionDirectors, String userEmail) {
```

Comment:
This shouldn't in a mapper. A mapper is all about the creation of an object, not for calculations. This should be in a service class